packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -87,11 +95,17 @@ export function FocusScope(props: FocusScopeProps) {
     }
 
     scopeRef.current = nodes;
-    scopes.add(scopeRef);
+  }, [children, parentScope]);
+
+  useLayoutEffect(() => {
+    scopes.set(scopeRef, parentScope);
     return () => {
+      if (activeScope === scopeRef) {
+        activeScope = parentScope;
+      }
```

Comment:
I think https://github.com/adobe/react-spectrum/pull/2223#issuecomment-901343206 might be coming from cleanup here
useLayoutEffect will run in the parent, then the child, so the activescope won't get set correctly